### DISCUSSION TITLE
Enhancing Sorting Functionality in DataTable for Rows with Empty Data

### DISCUSSION THREAD

**Discussion by MuhammedFaizel** [primefaces/community#1076] [PrimeReact] [answered]
In our current implementation, we've observed that the sorting functionality in the DataTable component is not behaving as expected when some rows contain empty data. Specifically, when sorting is applied, only rows with valid data are sorted and displayed, while rows with empty data are not prioritized or handled properly.

To address this issue, we need to enhance the sorting functionality to ensure that rows with empty data are also considered during sorting. Our goal is to have rows with empty data displayed at the top of the table when sorting is initiated. Upon subsequent sorting attempts, rows with valid data should be sorted and displayed on top, while rows with empty data should be positioned below.

This enhancement will provide a more intuitive and consistent sorting experience for users interacting with the DataTable component. We'll need to review our current sorting logic and make necessary adjustments to accommodate rows with empty data effectively, thereby improving the overall usability and functionality of our application's data display features.

**Comment by melloware** [comment, marked as answer]
@MuhammedFaizel isn't this setting what you want: https://primereact.org/configuration/#nullsortorder

It allows you to determine how empty cells are treated. Some people like it like Excel some people do not.

You want `-1` for your behavior 

> Determines how null values are sorted. The default value of 1 means sort like Excel with all NULL values at the bottom of the list. A value of -1 sorts NULL at the top of the list in ascending mode and at the bottom of the list in descending mode.

**Reply by MuhammedFaizel** [reply]
Hi,

interface IDataGrid {
	gridList?: any;
	filter?: any;
	collectorSummary?: any;
	onShowMap: (item: any) => void;
	onViewSummary?: (rowData: any) => void;
	onEditCollector?: (rowData: any) => void;
	onViewTeamMember?: (rowData: any) => void;
	onShowTrackedAssets?: (rowData: any) => void;
	onManageTeamMemberClicked?:(rowData: any) => void;
}
const CollectorsGrid: React.FC<IDataGrid> = (props: any) => {
	const { 
		gridList,
		filter,
		collectorSummary,
		onShowMap = () => {},
		onViewSummary = () => {},
		onEditCollector = () => {},
		onShowTrackedAssets = () => {},
		onManageTeamMemberClicked = () => {}
	} = props;
	
	const [teamMembers, showTeamMembers] = useState(false);
	const [sessionId, setSessionId] = useState<any>();
	const [collectorData, setCollectorData] = useState([]);
	
	//For Collector Summary
	const [collectorModalShow, setCollectorModalShow] = useState<any>(false);
	//For Collector delete propmt
	const [showDeleteCollectorPrompt, setShowDeleteCollectorPrompt] = useState(false);


	//redux
	const dispatch = useDispatch<any>();
	const notifications = useAppSelector((state) => state.uicontroller.notificationLists);
	const collectorDeletedDatas = useAppSelector((state) => state.collector.collectorDeletedData);

	useOnClickOutside(ref, () => showTeamMembers(false));
	
	useEffect(() => {
		if (gridList) {
			setCollectorData(gridList);
		}
	}, [gridList]);
	
	const sessionName = (rowData: any) => {
		return (
			<div className="grid-item-col d-flex align-items-center">
				<span className="name"><TextOverflow text={rowData?.["sessionName"]} /></span>
			</div>
		);
	};

	// Asset Data
	const totalAssetTrackedInfo = (rowData: any) => {
		return (
			<div className="grid-item-col d-flex align-items-center">
					<span className="name">
					<label onClick={() => onShowTrackedAssets(rowData?.sessionIdentifier)}>{rowData?.["trackedAssets"]}</label>
				</span>
			</div>
		);
	};

	const targetedAssetInfo = (rowData: any) => {
		return (
			<div className="grid-item-col d-flex align-items-center">
					<span className="name">
				<label>{rowData?.["targetedAssets"]}</label>
				</span>
			</div>
		);
	};


	//Description
	const descriptionTemplate = (rowData: any) => {
		return (
			<div className="grid-item-col d-flex align-items-center">
				<span id={`span${rowData?.["sessionIdentifier"]}`} className="description">
					<TextOverflow text={rowData["description"]} />
				</span>
			</div>
		);
	};

	const teamInfo = (rowData: any) => {
		const userNames: any = rowData?.["team"]?.slice(1).map((element: any) => element?.name).join(", ");
		const userName: any = rowData?.["team"]?.map((element: any) => element?.name).join(", ");
		return (
			<div className="grid-item-col d-flex align-items-center">
				{rowData?.["team"]?.length ? (
					rowData?.["team"]?.length === 1 ? (
						<TextOverflow text={rowData?.["team"][0]?.name} />
					) : rowData?.["team"]?.length === 2 ? (
						<>
							<TextOverflow text={userName} />
						</>
						) : (
							<>
								<div className="description">
									<div className="text-overflow">
										<TextOverflow text={rowData?.["team"][0]?.name} />
									</div>
								</div>
								<OverlayTrigger
									placement="top"
									overlay={
										<Tooltip id={`tooltip-${rowData["sessionIdentifier"]}`}>
											{userNames}
										</Tooltip>
									}
								>
									<div className="description">
										<div className="text-overflow">
											<a onClick={() => teamtoggleButton(rowData?.sessionIdentifier)} className="link">
												and&nbsp;{rowData?.["team"]?.length - 1}&nbsp;others
											</a>
										</div>
									</div>
								</OverlayTrigger>
							</>
					)
				) : (
					<TextOverflow text={"No Users selected"} />
				)}
			</div>
		);
	};
	  
	const teamtoggleButton = (id: any) => {
		showTeamMembers(!teamMembers);
		setSessionId(id);
	};

	//Status
	const status = (rowData: any) => {
		var statusName = "";
		if (rowData.status?.name) {
			statusName = rowData["status"]["name"].toLowerCase().split(" ").join(""); //to provide class name
		}

		return (
			<div className="grid-item-col d-flex align-items-center">
				<span className={"status " + statusName}>{rowData.status?.name}</span>
			</div>
		);
	};

	//Action Buttons
	const showMap = (item: any) => {
		onShowMap(item);
	};

	const onDeleteCollectorClick = (rowData: any) => {
		setShowDeleteCollectorPrompt(true);
		setSessionId(rowData);
	};

	const onDeleteCollector = () => {
		dispatch(onCollectorDelete(sessionId?.sessionName,sessionId?.sessionIdentifier, notifications));
	};

	const actionButton = (rowData: any) => {
		return (
			<div className="grid-item-col d-flex align-items-center">
				<div className="actions">
					<Button onClick={() => showMap(rowData)} variant="primary clear icon-only map-icon" title="Map">
						<MdLocationOn />
					</Button>
					<div className="more-dropdown">
						<Dropdown className="more-btn" drop="down">
							<DropdownToggle variant="base clear icon-only" title="More Option">
								<MdMoreVert />
							</DropdownToggle>
							<Dropdown.Menu>
								<Dropdown.Item onClick={() => onViewCollectorSummary(rowData)}>Summary</Dropdown.Item>
								<Dropdown.Item onClick={() => dispatch(copyCollectorEntities(rowData.sessionIdentifier))}>Copy Assets</Dropdown.Item>
								<Dropdown.Item onClick={() => onEditCollector(rowData?.sessionIdentifier)}>Edit Collector</Dropdown.Item>
								<Dropdown.Item onClick={() => onManageTeamMemberClicked(rowData)}>
									Manage Team Members
								</Dropdown.Item>
								<Dropdown.Item onClick={() => onDeleteCollectorClick(rowData)}>Delete</Dropdown.Item>
							</Dropdown.Menu>
						</Dropdown>
					</div>
				</div>
			</div>
		);
	};

	//view collector summary
	const onViewCollectorSummary = (rowData: any) => {
		onViewSummary(rowData);
		setCollectorModalShow(true);
	};

	//For Modal Accordion
	function AccordionToggle({ children, eventKey, callback }: any) {
		const currentEventKey = useContext(AccordionContext);
		// const toggleOnClick = useAccordionToggle(eventKey, () => callback && callback(eventKey));
		const isCurrentEventKey = currentEventKey === eventKey;

		return (<></>
			// <div className={"accordion-toggle " + (isCurrentEventKey ? "open" : "")} onClick={toggleOnClick}>
			// 	<MdKeyboardArrowRight />
			// </div>
		);
	}

	const TimelineLabel = (item: any) => {
		return (
			<div className="timeline-label">
				<div className="label">{item?.day}</div>
			</div>
		);
	};

	const TimelineData = (item: any) => {
		return (
			<div className="timeline-data">
				<div className="data">{item?.items}</div>
			</div>
		);
	};

	const plannedStartDate = (rowData: any) => {
		return (
			<div className="grid-item-col d-flex align-items-center">
				<span className="plannedDateInfo">
				{rowData?.plannedStartDate}
				</span>
			</div>
		);
	};
	const plannedEndDate = (rowData: any) => {
		return (
			<div className="grid-item-col d-flex align-items-center">
				<span className="plannedDateInfo">
					{rowData?.plannedEndDate}
				</span>
			</div>
		);
	};
	
	
	return (
		<>
			<DataTable value={collectorData} className="datagrid-template" globalFilter={filter} emptyMessage="" columnResizeMode="expand" resizableColumns >
				<Column header="Session Name" field="sessionName" body={sessionName} filterField="sessionName"  sortable={true}></Column>
				<Column header="Description" body={descriptionTemplate} field="description" filterField="description"  sortable={true}></Column>
				<Column header="Total Asset Tracked" body={totalAssetTrackedInfo}  field="trackedAssets" align="right" filterField="trackedAssets"  sortable={true}></Column>	
				<Column header="Targeted Asset" body={targetedAssetInfo}  field="targetedAssets" align="right" filterField="targetedAssets"  sortable={true}></Column>
				<Column header="Planned Start Date" body={plannedStartDate} field="plannedStartDate" filterField="plannedStartDate" sortable={true}></Column>
				<Column header="Planned End Date" body={plannedEndDate} field="plannedEndDate" filterField="plannedEndDate " sortable={true}></Column>
				<Column header="Team Members" body={teamInfo} filterField="team" field="team"  sortable={true} style={{ width: "300px" }}></Column>
				<Column header="Status" body={status} field="status.name" filterField="status.name"  sortable={true} style={{ width: "112px" }}></Column>
				<Column header="" body={actionButton}style={{ width: "96px" }}></Column>
			</DataTable>
				</>
	);
};

export default memo(CollectorsGrid);




response:-
{
    "data": [
        {
            "sessionIdentifier": "6b875cf9-26ae-4437-ad03-34a7a4ac91b0",
            "sessionName": "value 1121224324",
            "description": "value 121212sdkjhfgfjrehefdgrerhjgfevbdmnfd,mfnvdfgvjkdfhgkjdkldfngjkefhgvjkrnekjrengtjhgn",
            "targetedAssets": 12434,
            "trackedAssets": 0,
            "team": [
                {
                    "id": "370d176b-23cd-4a78-afc5-3eeb9dd1c6f1",
                    "name": "aaa12 cc",
                    "profilePic": null,
                    "email": null
                },
                {
                    "id": "c97050f0-b1f3-40d1-90e1-2e9a7a8d072b",
                    "name": "aabc aab",
                    "profilePic": null,
                    "email": null
                }
            ],
            "status": {
                "id": "404a80c9-8f7e-4a30-b0d4-460a37665964",
                "code": "DELETED",
                "name": "Deleted"
            },
            "additionalInfo": null,
            "plannedStartDate": null,
            "plannedEndDate": "31 Jan 2024"
        },
        {
            "sessionIdentifier": "6c458ac3-eedf-4b5f-8625-0c656676cf06",
            "sessionName": "edit test",
            "description": "v",
            "targetedAssets": 3,
            "trackedAssets": 0,
            "team": [],
            "status": {
                "id": "404a80c9-8f7e-4a30-b0d4-460a37665964",
                "code": "DELETED",
                "name": "Deleted"
            },
            "additionalInfo": null,
            "plannedStartDate": null,
            "plannedEndDate": null
        },
        {
            "sessionIdentifier": "b6da0e7f-6964-4d80-8439-9e4ad79fbe2c",
            "sessionName": "Demoo",
            "description": "Test Demo Description",
            "targetedAssets": 523,
            "trackedAssets": 0,
            "team": [],
            "status": {
                "id": "39eb37d3-64eb-444e-977e-4fdd9584e184",
                "code": "DRAFT",
                "name": "Draft"
            },
            "additionalInfo": null,
            "plannedStartDate": "08 Nov 2021",
            "plannedEndDate": "16 Nov 2021"
        },
        {
            "sessionIdentifier": "6e913dd3-4172-40fc-967d-877acec5ecc9",
            "sessionName": "72",
            "description": "72",
            "targetedAssets": 173,
            "trackedAssets": 0,
            "team": [],
            "status": {
                "id": "39eb37d3-64eb-444e-977e-4fdd9584e184",
                "code": "DRAFT",
                "name": "Draft"
            },
            "additionalInfo": null,
            "plannedStartDate": null,
            "plannedEndDate": null
        },
        {
            "sessionIdentifier": "8f378911-c3c7-4094-ae06-b7eb8cd474de",
            "sessionName": "232",
            "description": "67232",
            "targetedAssets": 173,
            "trackedAssets": 0,
            "team": [],
            "status": {
                "id": "39eb37d3-64eb-444e-977e-4fdd9584e184",
                "code": "DRAFT",
                "name": "Draft"
            },
            "additionalInfo": null,
            "plannedStartDate": null,
            "plannedEndDate": null
        },
        {
            "sessionIdentifier": "206c1130-292c-4037-b95d-7b58e23f4f16",
            "sessionName": "Abc",
            "description": "collector Description",
            "targetedAssets": 123,
            "trackedAssets": 0,
            "team": [],
            "status": {
                "id": "39eb37d3-64eb-444e-977e-4fdd9584e184",
                "code": "DRAFT",
                "name": "Draft"
            },
            "additionalInfo": null,
            "plannedStartDate": "08 Nov 2021",
            "plannedEndDate": "10 Nov 2021"
        },
        {
            "sessionIdentifier": "09794f91-65c4-43c2-8347-1aa4ef5d39dc",
            "sessionName": "Collector",
            "description": "sample Description",
            "targetedAssets": 1,
            "trackedAssets": 0,
            "team": [],
            "status": {
                "id": "39eb37d3-64eb-444e-977e-4fdd9584e184",
                "code": "DRAFT",
                "name": "Draft"
            },
            "additionalInfo": null,
            "plannedStartDate": "08 Nov 2021",
            "plannedEndDate": "12 Nov 2021"
        },
        {
            "sessionIdentifier": "3f16670b-0ee0-45ae-ab03-840f75a0f45e",
            "sessionName": "Team Collector",
            "description": "hhvjkhgnbmn bd Description",
            "targetedAssets": 52,
            "trackedAssets": 0,
            "team": [],
            "status": {
                "id": "39eb37d3-64eb-444e-977e-4fdd9584e184",
                "code": "DRAFT",
                "name": "Draft"
            },
            "additionalInfo": null,
            "plannedStartDate": "08 Nov 2021",
            "plannedEndDate": "13 Nov 2021"
        },
        {
            "sessionIdentifier": "686ccde3-54b5-444a-8159-1e9dfc72b167",
            "sessionName": "Team Collector 1",
            "description": "hhvjkhgnbmn bd Description",
            "targetedAssets": 52,
            "trackedAssets": 0,
            "team": [],
            "status": {
                "id": "39eb37d3-64eb-444e-977e-4fdd9584e184",
                "code": "DRAFT",
                "name": "Draft"
            },
            "additionalInfo": null,
            "plannedStartDate": "08 Nov 2021",
            "plannedEndDate": "13 Nov 2021"
        },
        {
            "sessionIdentifier": "990e3373-5e0a-4026-8cd0-8760ee9415d0",
            "sessionName": "Tesst Collector",
            "description": "hhvjkhgnbmn final Description",
            "targetedAssets": 15,
            "trackedAssets": 0,
            "team": [],
            "status": {
                "id": "39eb37d3-64eb-444e-977e-4fdd9584e184",
                "code": "DRAFT",
                "name": "Draft"
            },
            "additionalInfo": null,
            "plannedStartDate": "15 Nov 2021",
            "plannedEndDate": null
        },
        {
            "sessionIdentifier": "dbaeb3d7-9032-48c4-8fc4-2035193a6f26",
            "sessionName": "Session 12",
            "description": "Testt",
            "targetedAssets": 13,
            "trackedAssets": 0,
            "team": [],
            "status": {
                "id": "39eb37d3-64eb-444e-977e-4fdd9584e184",
                "code": "DRAFT",
                "name": "Draft"
            },
            "additionalInfo": null,
            "plannedStartDate": "20 May 2022",
            "plannedEndDate": null
        },
        {
            "sessionIdentifier": "d6c57670-1401-48da-a100-55f6ad23bd48",
            "sessionName": "fsf",
            "description": "gdfgdf",
            "targetedAssets": 44,
            "trackedAssets": 0,
            "team": [],
            "status": {
                "id": "39eb37d3-64eb-444e-977e-4fdd9584e184",
                "code": "DRAFT",
                "name": "Draft"
            },
            "additionalInfo": null,
            "plannedStartDate": "16 Jun 2023",
            "plannedEndDate": "22 Jun 2023"
        },
        {
            "sessionIdentifier": "380365e3-b607-4e53-84a5-83ed711efd9e",
            "sessionName": "adfsdf",
            "description": "fsdff",
            "targetedAssets": 55,
            "trackedAssets": 0,
            "team": [],
            "status": {
                "id": "39eb37d3-64eb-444e-977e-4fdd9584e184",
                "code": "DRAFT",
                "name": "Draft"
            },
            "additionalInfo": null,
            "plannedStartDate": "23 Jun 2023",
            "plannedEndDate": "30 Jun 2023"
        },
        {
            "sessionIdentifier": "2431e08a-c809-428f-8c1b-67eb238e06ce",
            "sessionName": "sdfsfs",
            "description": "sfsdfs",
            "targetedAssets": 66,
            "trackedAssets": 0,
            "team": [],
            "status": {
                "id": "39eb37d3-64eb-444e-977e-4fdd9584e184",
                "code": "DRAFT",
                "name": "Draft"
            },
            "additionalInfo": null,
            "plannedStartDate": "07 Jun 2023",
            "plannedEndDate": "21 Jun 2023"
        },
        {
            "sessionIdentifier": "d53c2e8f-fa2d-45c3-903d-7541c8870cbf",
            "sessionName": "fgsgsgsgs",
            "description": "gsgsdfs",
            "targetedAssets": 555,
            "trackedAssets": 0,
            "team": [],
            "status": {
                "id": "39eb37d3-64eb-444e-977e-4fdd9584e184",
                "code": "DRAFT",
                "name": "Draft"
            },
            "additionalInfo": null,
            "plannedStartDate": "19 Jun 2023",
            "plannedEndDate": "21 Jun 2023"
        },
        {
            "sessionIdentifier": "5ad54f41-ec95-4c30-b8c2-004b288c593d",
            "sessionName": "dsd",
            "description": "fsdfd",
            "targetedAssets": 44,
            "trackedAssets": 0,
            "team": [],
            "status": {
                "id": "39eb37d3-64eb-444e-977e-4fdd9584e184",
                "code": "DRAFT",
                "name": "Draft"
            },
            "additionalInfo": null,
            "plannedStartDate": null,
            "plannedEndDate": null
        },
        {
            "sessionIdentifier": "8f9e1b11-d7cb-4c57-bac2-b31112d34014",
            "sessionName": "54try",
            "description": "fdg",
            "targetedAssets": 55,
            "trackedAssets": 0,
            "team": [],
            "status": {
                "id": "39eb37d3-64eb-444e-977e-4fdd9584e184",
                "code": "DRAFT",
                "name": "Draft"
            },
            "additionalInfo": null,
            "plannedStartDate": null,
            "plannedEndDate": null
        },
        {
            "sessionIdentifier": "7668c17b-182c-4b02-a74b-469c7fcd5a3a",
            "sessionName": "r3rwerwe",
            "description": "",
            "targetedAssets": 8,
            "trackedAssets": 0,
            "team": [],
            "status": {
                "id": "39eb37d3-64eb-444e-977e-4fdd9584e184",
                "code": "DRAFT",
                "name": "Draft"
            },
            "additionalInfo": null,
            "plannedStartDate": null,
            "plannedEndDate": null
        },
        {
            "sessionIdentifier": "e35ff1a0-55b1-4a3b-8ae2-282ef7cc2000",
            "sessionName": "aedfse",
            "description": "",
            "targetedAssets": 5,
            "trackedAssets": 0,
            "team": [],
            "status": {
                "id": "39eb37d3-64eb-444e-977e-4fdd9584e184",
                "code": "DRAFT",
                "name": "Draft"
            },
            "additionalInfo": null,
            "plannedStartDate": null,
            "plannedEndDate": null
        },
        {
            "sessionIdentifier": "cd7de2d0-6291-427f-b1b6-6024aa77831a",
            "sessionName": "safds",
            "description": "",
            "targetedAssets": 12,
            "trackedAssets": 0,
            "team": [],
            "status": {
                "id": "39eb37d3-64eb-444e-977e-4fdd9584e184",
                "code": "DRAFT",
                "name": "Draft"
            },
            "additionalInfo": null,
            "plannedStartDate": null,
            "plannedEndDate": null
        },
        {
            "sessionIdentifier": "80a0957f-3224-47ba-b6be-af7a8daef3a3",
            "sessionName": "qw",
            "description": null,
            "targetedAssets": 23,
            "trackedAssets": 0,
            "team": [],
            "status": {
                "id": "39eb37d3-64eb-444e-977e-4fdd9584e184",
                "code": "DRAFT",
                "name": "Draft"
            },
            "additionalInfo": null,
            "plannedStartDate": null,
            "plannedEndDate": null
        },
        {
            "sessionIdentifier": "d39b04f1-3951-44e9-8018-081a0eb127c2",
            "sessionName": "weq",
            "description": null,
            "targetedAssets": 32,
            "trackedAssets": 0,
            "team": [],
            "status": {
                "id": "39eb37d3-64eb-444e-977e-4fdd9584e184",
                "code": "DRAFT",
                "name": "Draft"
            },
            "additionalInfo": null,
            "plannedStartDate": null,
            "plannedEndDate": null
        },
        {
            "sessionIdentifier": "1284d641-ea46-4719-85e4-bd19c67de571",
            "sessionName": "Session Name 12",
            "description": null,
            "targetedAssets": 34,
            "trackedAssets": 0,
            "team": [],
            "status": {
                "id": "39eb37d3-64eb-444e-977e-4fdd9584e184",
                "code": "DRAFT",
                "name": "Draft"
            },
            "additionalInfo": null,
            "plannedStartDate": null,
            "plannedEndDate": null
        },
        {
            "sessionIdentifier": "c1d659b9-01a3-425c-89e5-4d681cd93f79",
            "sessionName": "2w2w2",
            "description": null,
            "targetedAssets": 2212,
            "trackedAssets": 0,
            "team": [],
            "status": {
                "id": "39eb37d3-64eb-444e-977e-4fdd9584e184",
                "code": "DRAFT",
                "name": "Draft"
            },
            "additionalInfo": null,
            "plannedStartDate": null,
            "plannedEndDate": null
        },
        {
            "sessionIdentifier": "9cb8dc21-fecc-4a09-9827-008b84f7217c",
            "sessionName": "werty",
            "description": null,
            "targetedAssets": 76543,
            "trackedAssets": 0,
            "team": [],
            "status": {
                "id": "39eb37d3-64eb-444e-977e-4fdd9584e184",
                "code": "DRAFT",
                "name": "Draft"
            },
            "additionalInfo": null,
            "plannedStartDate": null,
            "plannedEndDate": null
        },
        {
            "sessionIdentifier": "4b025f41-6be5-424f-ac46-c2dd7b137341",
            "sessionName": "wqqdwsmdhgwdwdedvhjjfhejegjejhfgejhgfdwdfsdfsdjgfshgfsgfjsdfgejhgfjehrgfjrehgfje,rhbf,jrhgfjerbfkjre",
            "description": "wdhgqwhdwehwhewefhdedgvjbhdcnsdnsvgjhefgewgfwjgfwejhgfwehjwegwdhgqwhdwehwhewefhdedgvjbhdcnsdnsvgjhefgewgfwjgfwejhgfwehjwegwdhgqwhdwehwhewefhdedgvjbhdcnsdnsvgjhefgewgfwjgfwejhgfwehjwegwdhgqwhdwehwhewefhdedgvjbhdcnsdnsvgjhefgewgfwjgfwejhgfwehjwegwdhgqwhdwehwhewefhdedgvjbhdcnsdnsvgjhefgewgfwjgfwejhgfwehjwegwdhgqwhdwehwhewefhdedgvjbhdcnsdnsvgjhefgewgfwjgfwejhgfwehjwegwdhgqwhdwehwhewefhdedgvjbhdcnsdnsvgjhefgewgfwjgfwejhgfwehjwegwdhgqwhdwehwhewefhdedgvjbhdcnsdnsvgjhefgewgfwjgfwejhgfwehjwegwdhgqwhdwehwhewefhdedgvjbhdcnsdnsvgjhefgewgfwjgfwejhgfwehjwegwdhgqwhdwehwhewefhdedgvjbhdcnsdnsvgjhefgewgfwjgfwejhgfwehjwegwdhgqwhdwehwhewefhdedgvjbhdcnsdnsvgjhefgewgfwjgfwejhgfwehjwegwdhgqwhdwehwhewefhdedgvjbhdcnsdnsvgjhefgewgfwjgfwejhgfwehjwegwdhgqwhdwehwhewefhdedgvjbhdcnsdnsvgjhefgewgfwjgfwejhgfwehjwegwdhgqwhdwehwhewefhdedgvjbhdcnsdnsvgjhefgewgfwjgfwejhgfwehjwegwdhgqwhdwehwhewefhdedgvjbhdcnsdnsvgjhefgewgfwjgfwejhgfwehjwegwdhgqwhdwehwhewefhdedgvjbhdcnsdnsvgjhefgewgfwjgfwejhgfwehjwegwdhgqwhdwehwhewefhdedgvj",
            "targetedAssets": 2,
            "trackedAssets": 0,
            "team": [],
            "status": {
                "id": "39eb37d3-64eb-444e-977e-4fdd9584e184",
                "code": "DRAFT",
                "name": "Draft"
            },
            "additionalInfo": null,
            "plannedStartDate": "18 Jan 2024",
            "plannedEndDate": "24 Jan 2024"
        },
        {
            "sessionIdentifier": "84860eba-302c-4589-80b3-0b09b68327c8",
            "sessionName": "Session",
            "description": null,
            "targetedAssets": 7,
            "trackedAssets": 0,
            "team": [],
            "status": {
                "id": "39eb37d3-64eb-444e-977e-4fdd9584e184",
                "code": "DRAFT",
                "name": "Draft"
            },
            "additionalInfo": null,
            "plannedStartDate": null,
            "plannedEndDate": null
        },
        {
            "sessionIdentifier": "302bbca4-f80a-4449-b6b9-11ce32af7fcc",
            "sessionName": "test collector",
            "description": null,
            "targetedAssets": 55,
            "trackedAssets": 0,
            "team": [],
            "status": {
                "id": "39eb37d3-64eb-444e-977e-4fdd9584e184",
                "code": "DRAFT",
                "name": "Draft"
            },
            "additionalInfo": null,
            "plannedStartDate": null,
            "plannedEndDate": null
        },
        {
            "sessionIdentifier": "a7f101ff-0795-4f68-9e25-8795788c956f",
            "sessionName": "Test 22",
            "description": null,
            "targetedAssets": 21,
            "trackedAssets": 0,
            "team": [],
            "status": {
                "id": "39eb37d3-64eb-444e-977e-4fdd9584e184",
                "code": "DRAFT",
                "name": "Draft"
            },
            "additionalInfo": null,
            "plannedStartDate": null,
            "plannedEndDate": "31 Jan 2024"
        },
        {
            "sessionIdentifier": "63790904-fea8-4d49-8f19-4baccaec4dab",
            "sessionName": "dd",
            "description": null,
            "targetedAssets": 33,
            "trackedAssets": 0,
            "team": [],
            "status": {
                "id": "39eb37d3-64eb-444e-977e-4fdd9584e184",
                "code": "DRAFT",
                "name": "Draft"
            },
            "additionalInfo": null,
            "plannedStartDate": null,
            "plannedEndDate": null
        }
    ],
    "message": null,
    "isError": false,
    "messageCode": 200,
    "status": "done"
}

How can we add the null sorting functinality??

**Comment by melloware** [comment]
Please create a Stackblitz reproducer. Pasting all this code does no one any good.  And how to set `nullSortOrder` is in the link I posted: https://primereact.org/configuration/#nullsortorder
